client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -1,13 +1,20 @@
+import { dispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
 import { translate } from 'i18n-calypso';
 import { PLANS_LIST } from 'calypso/../packages/calypso-products/src/plans-list';
 import { SiteDetails } from 'calypso/../packages/data-stores/src';
+import { NavigationControls } from 'calypso/landing/stepper/declarative-flow/internals/types';
+import { SITE_STORE } from '../../../../stores';
 import { launchpadFlowTasks } from './tasks';
 import { Task } from './types';
 
 export function getEnhancedTasks(
 	tasks: Task[],
 	siteSlug: string | null,
-	site: SiteDetails | null
+	site: SiteDetails | null,
+	submit: NavigationControls[ 'submit' ],
+	setPendingAction: ( pendingAction: ( () => Promise< void > ) | undefined ) => void,
+	setProgressTitle: ( progressTitle: string | undefined ) => void
```

Comment:
Not that we are necessarily doing anything wrong, but Im curious if there is advantage to what we are doing in terms of getting these actions from dispatch in `checklist.tsx`, passing them to this function, assigning types here, etc. as opposed to just using `dispatch( ONBOARD_STORE )` down in the case that uses these below?

Reply:
> Not that we are necessarily doing anything wrong, but Im curious if there is advantage to what we are doing in terms of getting these actions from dispatch in checklist.tsx, passing them to this function, assigning types here, etc. as opposed to just using dispatch( ONBOARD_STORE ) down in the case that uses these below?

Good catch 👍 

I was caught up in the prop drilling of the `submit` method, but `setPendingAction` and `setProgressTitle` can be defined in `task-helper` itself